### PR TITLE
Restore demog functionality. Fix n core adjustment. Add in ML imps for observation-level data.

### PR DESCRIPTION
**Pull request by conig**
I was super clever and managed to break the pipeline with #69.
This fixes that, also I had my `>` the wrong way round.

### CODE REVIEW

File: R/make_data_imp.R [hunk 1]
```diff
@@ -12,7 +12,8 @@
 
 make_data_imp <- function(data, n_imps = 3) {
   data <- data %>%
-    select(-n_valid_hours, -n_hours, -weekday_x, -day_zero)
+    select(-n_valid_hours, -n_hours, -weekday_x, -day_zero) |>
+    dplyr::mutate(participant_id = as.integer(factor(participant_id)))
```
Comment (tarensanders) [inline R/make_data_imp.R:16]: Didn't you already make this change in the clean_data script?

File: R/make_data_imp.R [hunk 2]
```diff
@@ -31,7 +32,7 @@ make_data_imp <- function(data, n_imps = 3) {
 
   # Run imps with better settings
   future_cores <- parallel::detectCores() - 1
-  if(future_cores < n_imps) future_cores <- n_imps
+  if(future_cores > n_imps) future_cores <- n_imps
```
Comment (tarensanders) [inline R/make_data_imp.R]: What about making this simpler:

```suggestion
future_cores <- min(parallel::detectCores() - 1, n_imps)
```

File: R/clean_data.R [hunk 3]
```diff
@@ -269,7 +269,5 @@ clean_data <- function(data_joined) {
     )) %>%
     ungroup()
 
-  d$participant_id <- as.integer(factor(d$participant_id))
```
Comment (tarensanders) [inline R/clean_data.R:272]: Maybe just add this as part of line 50, rather than as an add-on?

**Review by tarensanders — COMMENTED** [review]
Looks good - and can confirm it runs (should have tried that last time). See minor suggestions

File: R/clean_data.R (hunk 3 above)
Comment (tarensanders) [reply R/clean_data.R:272]: Wait, sorry this is a deletion. My bad. Why does it need to be in the imp and not the cleaning script?

Comment (conig) [reply R/clean_data.R:272]: Demog expects it to be a factor and dumps it when it's an integer. So I think makes sense just to have it as an integer when making the imps (which is needed for the multilevel function to work).

File: R/make_data_imp.R (hunk 2 above)
Comment (conig) [reply R/make_data_imp.R]: Looks good, weirdly wouldn't let me commit this suggestion so have pushed it manually.

**Comment by conig** [comment]
Putting in draft as we may want to do further multi-level imputation.

**Pull request merged**